Gradle support for agent does not work.

Running 

```
./gradlew -Pagent test   
```

followed by 

```
./gradlew -Pagent nativeTest
```

does not work but ends with

```
> Task :nativeTestBuild FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTestBuild'.
> Agent output missing while `agent` option is set.
  Did you run the test task before with `-Pagent` enabled?

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org
```

even though native files are generated:

```
 tree build/native 
build/native
└── agent-output
    └── test
        ├── jni-config.json
        ├── proxy-config.json
        ├── reflect-config.json
        ├── resource-config.json
        └── serialization-config.json
```

however the plugin tries to resolve https://github.com/graalvm/native-build-tools/blob/a28d187f43446493fcba78c7e372cd3bd31e3004/native-gradle-plugin/src/main/java/org/graalvm/buildtools/Utils.java#L61 with 

https://github.com/graalvm/native-build-tools/blob/a28d187f43446493fcba78c7e372cd3bd31e3004/native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/GradleUtils.java#L105

which is wrong in two places:

The `test` path is not in there but the native path will be resolved twice.

Moving the agent files into 

```
tree build/native/native 
build/native/native
└── agent-output
    └── test
        ├── jni-config.json
        ├── proxy-config.json
        ├── reflect-config.json
        ├── resource-config.json
        └── serialization-config.json
```

"fixes" this…

--- a/native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/dsl/NativeImageOptions.java
+++ b/native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/dsl/NativeImageOptions.java
@@ -175,8 +175,8 @@ public class NativeImageOptions {
         }
 
         if (project.hasProperty(Utils.AGENT_PROPERTY) || getAgent().get()) {
-            Path agentOutput = GradleUtils.getTargetDir(project)
-                    .resolve(Utils.AGENT_OUTPUT_FOLDER).toAbsolutePath();
+            Path agentOutput = project.getBuildDir().toPath()
+                    .resolve(Utils.AGENT_OUTPUT_FOLDER).resolve(sourceSetName).toAbsolutePath();
 
             if (!agentOutput.toFile().exists()) {
                 // Maybe user chose to persist configuration into the codebase, so lets also check if that folder exists.

